Description: Use ruby-ronn and marked to generate documentation
 Command-line options are different. Perl doesn't need -i switch.
Forwarded: not-needed
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2014-07-29
Index: npm/scripts/doc-build.sh
===================================================================
--- npm.orig/scripts/doc-build.sh
+++ npm/scripts/doc-build.sh
@@ -1,63 +1,5 @@
 #!/usr/bin/env bash

-if [[ $DEBUG != "" ]]; then
-  set -x
-fi
-set -o errexit
-set -o pipefail
-
-if ! [ -x node_modules/.bin/marked-man ]; then
-  ps=0
-  if [ -f .building_marked-man ]; then
-    pid=$(cat .building_marked-man)
-    ps=$(ps -p $pid | grep $pid | wc -l) || true
-  fi
-
-  if [ -f .building_marked-man ] && [ $ps != 0 ]; then
-    while [ -f .building_marked-man ]; do
-      sleep 1
-    done
-  else
-    # a race to see which make process will be the one to install marked-man
-    echo $$ > .building_marked-man
-    sleep 1
-    if [ $(cat .building_marked-man) == $$ ]; then
-      make node_modules/.bin/marked-man
-      rm .building_marked-man
-    else
-      while [ -f .building_marked-man ]; do
-        sleep 1
-      done
-    fi
-  fi
-fi
-
-if ! [ -x node_modules/.bin/marked ]; then
-  ps=0
-  if [ -f .building_marked ]; then
-    pid=$(cat .building_marked)
-    ps=$(ps -p $pid | grep $pid | wc -l) || true
-  fi
-
-  if [ -f .building_marked ] && [ $ps != 0 ]; then
-    while [ -f .building_marked ]; do
-      sleep 1
-    done
-  else
-    # a race to see which make process will be the one to install marked
-    echo $$ > .building_marked
-    sleep 1
-    if [ $(cat .building_marked) == $$ ]; then
-      make node_modules/.bin/marked
-      rm .building_marked
-    else
-      while [ -f .building_marked ]; do
-        sleep 1
-      done
-    fi
-  fi
-fi
-
 src=$1
 dest=$2
 name=$(basename ${src%.*})
@@ -99,14 +41,14 @@ man_replace_tokens () {

 case $dest in
   *.[1357])
-    ./node_modules/.bin/marked-man --roff $src \
+    cat $src | ronn \
     | man_replace_tokens > $dest
     exit $?
     ;;
   *.html)
     url=${dest/html\//}
     (cat html/dochead.html && \
-     cat $src | ./node_modules/.bin/marked &&
+     cat $src | marked &&
      cat html/docfoot.html)\
     | html_replace_tokens $url \
     > $dest
